pkg/virt-controller/watch/workload-updater/workload-updater.go

```diff
@@ -294,7 +296,11 @@ func (c *WorkloadUpdateController) getUpdateData(kv *virtv1.KubeVirt) *updateDat
 
 	for _, method := range kv.Spec.WorkloadUpdateStrategy.WorkloadUpdateMethods {
 		if method == virtv1.WorkloadUpdateMethodLiveMigrate {
-			automatedMigrationAllowed = true
+			if c.clusterConfig.LiveMigrationEnabled() {
+				automatedMigrationAllowed = true
+			} else {
+				c.recorder.Eventf(kv, k8sv1.EventTypeWarning, FailedEnableWorkloadMigrationReason, "Error enabling live-migration method for workload updates: the '%s' feature-gate is not enabled", virtconfig.LiveMigrationGate)
```

Comment:
Do you know how often this will fire? Also, I am not sure it is the best place to fire the event as this is going to happen no matter how many migratable VMIs are in the cluster(including 0).

Reply:
1. It might be pretty often if not now then in the future. I would rather be on the safe side and have some rate limiter here but I will let others express opinions on this.

2. You are right!

Reply:
Another idea would be to have the alert fired in https://github.com/kubevirt/kubevirt/pull/7406, inside the webhook but that main concern is that it would slow down the webhook.

I'd probably leave it here without rate-limiting as the KubeVirt CR doesn't get updated that often, and we should be loud if there is a misconfiguration.

Reply:
@davidvossel What is your opinion on this? My argument is that there is no bound number of iterations that the controller can do.